UserAgent/Views/Privacy Indicator/PrivacyDashboardUtils.swift

```diff
@@ -1,125 +1,110 @@
+            super.init(frame: CGRect.zero)
+            let (dot, label) = callback()
+            self.columns(dot: dot, label: label)
         }
-        return stackView
-    }
-
-    static func dotView(_ color: UIColor) -> UIView {
-        let wrapper = self.dotWrapperView()
-        self.dotMainView(wrapper, color)
-        return wrapper
-    }
-
-    static func dotWrapperView() -> UIView {
-        let radius = 5
-        let margin = 4
-        let view = UIView()
-        view.snp.makeConstraints { make in
-            make.width.equalTo(radius * 2)
-            make.height.equalTo((radius * 2) + (margin * 2))
+        init(_ callback: () -> (UIView, UILabel, UILabel)) {
+            super.init(frame: CGRect.zero)
+            let (dot, label, number) = callback()
+            self.columns(dot: dot, label: label, number: number)
         }
-        return view
-    }
-
-    static func dotMainView(
-        _ wrapper: UIView,
-        _ color: UIColor
-    ) {
-        let radius = 5
-        let view = UIView()
-        view.backgroundColor = color
-        view.layer.cornerRadius = CGFloat(radius)
-        wrapper.addSubview(view)
+        private func columns(dot: UIView, label: UILabel, number: UILabel) {
+            self.spacing = 5
+            self.alignment = .top
+            self.distribution = .fill
+            [dot, label, number].forEach { self.addArrangedSubview($0)}
+            number.alpha = 0.7
 
-        view.snp.makeConstraints { make in
-            make.width.equalTo(view.snp.height)
-            make.height.equalTo(radius * 2)
-            make.center.equalTo(wrapper)
+            label.snp.makeConstraints { make in
+                make.width.lessThanOrEqualTo(100)
+            }
+            number.snp.makeConstraints { make in
+                // space for keeping three digits [18, 25]
+                make.width.greaterThanOrEqualTo(18)
+                make.width.lessThanOrEqualTo(25)
+            }
+        }
+        private func columns(dot: UIView, label: UILabel) {
+            self.spacing = 5
+            self.alignment = .top
+            [dot, label].forEach { self.addArrangedSubview($0)}
         }
     }
 
-    static func statLabel(_ text: String) -> UILabel {
-        let label = UILabel()
-        label.textColor = Theme.textField.textAndTint
-        label.text = text
-        label.numberOfLines = 0
-        label.setContentCompressionResistancePriority(.required, for: .horizontal)
-        label.font = UIFont.preferredFont(forTextStyle: .footnote)
-        return label
-    }
-
-    static func titleLabel(withStatus status: BlockerStatus?) -> UILabel {
-        let label = UILabel()
-        label.font = UIFont.preferredFont(forTextStyle: .title1)
-        label.adjustsFontForContentSizeCategory = true
-        label.numberOfLines = 3
-        label.lineBreakMode = .byWordWrapping
-        label.text = self.headerText(status)
-        return label
+    class Label: UILabel {
```

Comment:
By default the visibility of classes are marked as "Internal". Means it is visible with in the project. So naming classes like Label, Dot, etc. should serve to generic purpose, otherwise it will be misleading. Or you make mark those classes to private or fileprivate if you use it only with in the class. Or you can name them more specifically.